numpy/random/bit_generator.pyx

```diff
@@ -43,13 +43,12 @@ except ImportError:
     from random import SystemRandom
     randbits = SystemRandom().getrandbits
 
-try:
-    from threading import Lock
-except ImportError:
-    from dummy_threading import Lock
-
 from cpython.pycapsule cimport PyCapsule_New
 
+# Threading isn't the cheapest import. Import from the lightweight
+# _thread module,
+from _thread import allocate_lock as Lock
```

Comment:
It seems to me that the `_thread` method might be choking the pypy test? (i.e. that lock waits forever?). How OK is it really to use the `_thread` name?

Reply:
The python STL was documenting it.

But I guess I misunderstood what they meant by low level.

https://docs.python.org/3/library/concurrency.html

I kinda knew I was doing a bad thing, so I didn't try it first. I think I might have to rerun initial benchmarks now that some of the other PRs are going through

Reply:
Dunno, it seems that python might have started to endorse `_thread` specfically, and possibly pypy just has a bug.

Reply:
I have to start rerunning these benchmarks again. Some of these `fixes` are related to deprecation cleanups. Others like this one are more hazy. One could argue that importing from `_thread` is obscure and shouldn't be done even if it was supported by both PyPy.

Reply:
```
root@c766aae4e273:/# pypy3.6-v7.1.1-linux64/bin/pypy3 
Python 3.6.1 (784b254d6699, Apr 14 2019, 10:22:42)
[PyPy 7.1.1-beta0 with GCC 6.2.0 20160901] on linux
Type "help", "copyright", "credits" or "license" for more information.
And now for something completely different: ``snow, snow''
>>>> import _thread
>>>> from _thread import allocate_lock as Lock
>>>> Lock
<built-in function allocate_lock>
```
This seems to work on the latest downloaded PyPy